src/utils/scryfall.ts

```diff
@@ -80,7 +80,8 @@ export async function fetchCollection(
     typeLine: string
   }[]
 > {
-  let batch = cardNames.splice(0, scryfallFetchLimit)
+  const newLineRegex = /^\n$|^\r$|^\s{1,}$|^$/
+  let batch = cardNames.splice(0, scryfallFetchLimit).filter((name) => !name.match(newLineRegex))
```

Comment:
I'd flip the filtering and batching first, so (contrived example) cards spaced out with 74 lines of whitespace wouldn't result in one card per fetch, and store the cleaned up list in a variable above this loop since it also needs to be used. The filtered list should be used to get the next batch at the end of this loop too.

Just related to the fact that this wasn't obvious, is there a better way to write this loop so we don't have to call `splice` twice?

Reply:
How you plan to implement the paste in the future would probably change my answer. Ideally I'd want validation to warn the user that they can't paste a list longer than 75 before we actually start transforming the data.

Reply:
More than 75 cards works. They're just split into multiple requests.

Ultimately I would like the import feature to first present the pasted list of cards, giving the user an option to review and confirm before importing cards, and also support cards with counts and headings like 'main' and 'sideboard'. Not sure about the details, but might be good to make this query to Scryfall before presenting that dialog and showing any errors in there too.

Reply:
Ah, kinda just figured out what I think the batch logic is doing. I would probably use lodash's `chunk` function and loop through the result of that to construct the cardpool.